src/vs/workbench/parts/debug/common/debugModel.ts

```diff
@@ -90,16 +91,61 @@ export function getFullExpressionName(expression: debug.IExpression, sessionType
 }
 
 export class Thread implements debug.IThread {
-
+	private promisedCallStack: TPromise<debug.IStackFrame[]>;
+	private cachedCallStack: debug.IStackFrame[];
+	private callStackQueried: boolean;
 	public stoppedDetails: debug.IRawStoppedDetails;
+	public stopped: boolean;
 
-	constructor(public name: string, public threadId, public callStack: debug.IStackFrame[]) {
+	constructor(public name: string, public threadId) {
```

Comment:
Not important - but I personally am not a big fun of using `undefined`. I always try to use `null`, because having one `null` is already bad enough. You do not have to change anything here, but just for the future so you know that is my style.